src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/PublishReleaseFilesFunction.cs

```diff
@@ -57,7 +60,14 @@ public class PublishReleaseFilesFunction
                 await UpdateStage(releaseId, releaseStatusId, Started);
                 try
                 {
-                    _publishingService.PublishReleaseFiles(releaseId).Wait();
+                    var methodology = await _methodologyService.GetByRelease(releaseId);
+                    // Publish the files of the Methodology if it's not already live
+                    // Since the Methodology will be published for the first time with this Release
+                    if (methodology != null && !methodology.Live)
```

Comment:
I thought about this, and I'm not sure.

It would be inconsistent with the functionality already live, which is that methodology content is published already regardless of the status. This is just publishing the files to match that.

I think this is always going to be a problem until we have methodology amendments because there's nothing stopping anybody unapproving a methodology at any point currently.

The only check in place right now is that Methodology should be Approved at the time a Publication is created and when the Release is Approved because it's a Release checklist item.

We could add this check in but we'd have to exclude Publishing the methodology content too, and it might go unnoticed that the methodology hasn't been published.

If we do add these checks in though, we'll need to make a change to the Admins `MethodologyService` where it works out if the Methodology should be published immediately or not. At the moment if it's already live it publishes it immediately, otherwise wait for this first publication to be published instead. That logic will no longer work.